tests/draft2020-12/ref.json

```diff
@@ -883,5 +883,42 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "$ref must be resolved against the current base",
+        "schema": {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "$ref": "#/$defs/foo/$defs/b",
+            "$defs": {
+                "foo":{
+                    "$id": "http://localhost:1234/draft2020-12/ref/foo",
```

Comment:
Just FYI:  `http://localhost:1234/` urls are typically configured to look in the `remotes/` folder.